sympy/core/expr.py

```diff
@@ -2673,6 +2703,41 @@ def series(self, x=None, x0=0, n=6, dir="+", logx=None):
         x
         >>> abs(x).series(dir="-")
         -x
+        >>> f = tan(x)
+        >>> f.series(x, 2, 6, "+")
+        tan(2) + (1 + tan(2)**2)*(x - 2) + (x - 2)**2*(tan(2)**3 + tan(2)) +
+        (x - 2)**3*(1/3 + 4*tan(2)**2/3 + tan(2)**4) + (x - 2)**4*(tan(2)**5 +
+        5*tan(2)**3/3 + 2*tan(2)/3) + (x - 2)**5*(2/15 + 17*tan(2)**2/15 +
+        2*tan(2)**4 + tan(2)**6) + O((x - 2)**6, (x, 2))
+
+        >>> f.series(x, 2, 3, "-")
+        tan(2) + (2 - x)*(-tan(2)**2 - 1) + (2 - x)**2*(tan(2)**3 + tan(2))
+        + O((x - 2)**3, (x, 2))
+
+        >>> f.series(x, 2, oo, "+")
```

Comment:
Can this failed example be included in tests rather than in the docstrings?

Reply:
Shall I add it in tests and cancel from docstrings?

Reply:
From my point of view, the answer is Yes.

Reply:
> Shall I add it in tests and cancel from docstrings?

probably, no use of keeping two copies.
it is getting tested both ways.

Reply:
Tests for incorrect inputs should be avoided in docstrings.

Reply:
alright. How to check typeerror in tests? Shall I use decorator @XFAIL ?

Reply:
probably
```python
raises(TypeError, lambda: function)
```

Reply:
In my opinion, it you should make it documented in words if the the test is moved, 
like in `dir` argument, or you can make a new section `Raises` and add possible exceptions that can be thrown from the method.

Reply:
I like the idea of the Raises section
It can include every possible exception and why were the exceptions raised.
That might be helpful to someone who might not be completely sure why the function is raising an exception.
Although the error messages also include a description of the error but a well documented explanation of why it was raised and what can be done to avoid that can be helpful.

Reply:
Shall I add this ``raises `` section under def series in sympy/core/expr.py?

Reply:
IMO, Adding a single line like ``oo`` or ``-oo`` can't be an integer as under ``x0`` and ``n`` will do?

Reply:
@sylee957 @Sc0rpi0n101 is it good?